apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data.IntegrationTests/StructParameterTests.cs

```diff
@@ -0,0 +1,304 @@
+                    cmd.Parameters.Add("struct_param", outerStruct.GetSpannerDbType(), null);
+
+                    using (var reader = await cmd.ExecuteReaderAsync())
+                    {
+                        Assert.True(await reader.ReadAsync());
+                        Assert.True(reader.IsDBNull(0));
+                        Assert.False(await reader.ReadAsync());
+                    }
+                }
+            }
+        }
+
+        [Fact]
+        public async Task EmptyStruct()
+        {
+            var emptyStruct = new SpannerStruct();
+
+            using (var connection = await _fixture.GetTestDatabaseConnectionAsync())
+            {
+                using (var cmd = connection.CreateSelectCommand("SELECT @struct_param IS NULL"))
+                {
+                    cmd.Parameters.Add("struct_param", emptyStruct.GetSpannerDbType(), emptyStruct);
+
+                    using (var reader = await cmd.ExecuteReaderAsync())
+                    {
+                        Assert.True(await reader.ReadAsync());
+                        Assert.False(reader.GetBoolean(0));
+                        Assert.False(await reader.ReadAsync());
+                    }
+                }
+            }
+        }
+
+        [Fact]
+        public async Task EmptyStruct_NullValue()
+        {
+            var emptyStruct = new SpannerStruct();
+
+            using (var connection = await _fixture.GetTestDatabaseConnectionAsync())
+            {
+                using (var cmd = connection.CreateSelectCommand("SELECT @struct_param IS NULL"))
+                {
+                    cmd.Parameters.Add("struct_param", emptyStruct.GetSpannerDbType(), null);
+
+                    using (var reader = await cmd.ExecuteReaderAsync())
+                    {
+                        Assert.True(await reader.ReadAsync());
+                        Assert.True(reader.GetBoolean(0));
+                        Assert.False(await reader.ReadAsync());
+                    }
+                }
+            }
+        }
+
+        [Fact]
+        public async Task NullFieldValue()
+        {
+            var structValue = new SpannerStruct
+            {
+                { "f1", SpannerDbType.Int64, null }
+            };
+
+            using (var connection = await _fixture.GetTestDatabaseConnectionAsync())
+            {
+                using (var cmd = connection.CreateSelectCommand("SELECT @struct_param.f1"))
+                {
+                    cmd.Parameters.Add("struct_param", structValue.GetSpannerDbType(), structValue);
+
+                    using (var reader = await cmd.ExecuteReaderAsync())
+                    {
+                        Assert.True(await reader.ReadAsync());
+                        Assert.True(reader.IsDBNull(0));
+                        Assert.False(await reader.ReadAsync());
+                    }
+                }
+            }
+        }
+
+        [Fact]
+        public async Task ServerStructEquality()
```

Comment:
Maybe add an associated equality check using the wrong values to verify the values are actually being compared?

Reply:
Sure. (This was basically a list of suggested tests - happy to add another one though.)